narwhal/types/src/primary.rs

```diff
@@ -30,9 +31,61 @@ use std::{
 /// The round number.
 pub type Round = u64;
 
+/// The epoch UNIX timestamp in milliseconds
+pub type TimestampMs = u64;
+
+pub trait Timestamp {
+    fn elapsed(&self) -> Duration;
```

Comment:
This name was quite confusing to me at first? My line of thought was: how would the concept of "Timestamp" as a trait have a notion of "elapsed"? A timestamp is a single instant in time so I think it cannot have a duration except in relation to some other timestamp?

Can we use nomenclature compatible with std::time where possible? So maybe call this `duration_to_now`?

Reply:
Good point @aschran ! To be honest I followed the same convention that `Instant` uses https://doc.rust-lang.org/std/time/struct.Instant.html#method.elapsed of the `time` package. 

where Instant is also basically representing an "instance at a specific point in time" and the "elapsed" is measuring the time passed until "now". That being said I am definitely open to rename though

Reply:
Why do we need the trait? Do we have multiple structs implementing it?

Reply:
The `TimestampMs` is a `type` thus I am not allowed to add an implementation for it. I can only implement a `trait` for it (see line 41). That's the reason I introduced a trait